test/e2e/cluster_upgrade_test.go

```diff
@@ -51,3 +52,51 @@ var _ = Describe("When upgrading a workload cluster using ClusterClass", func()
 		}
 	})
 })
+
+var _ = Describe("When testing KCP upgrade in a single control plane cluster", func() {
```

Comment:
I've kept the test names the same for now. Should we rename them in something like:
* `When upgrading a workload cluster with a single control plane machine`
* `When upgrading a workload cluster with a HA control plane`
* `When upgrading a workload cluster with a HA control plane using scale-in rollout`

I think I'm in favor of renaming

Reply:
+1 for renaming